packages/cli/src/DaemonClient.ts

```diff
@@ -67,48 +56,79 @@ export class DaemonClient {
         Imperative.api.appLogger.trace('client closed');
     }
 
+    /**
+     * Shutdown the daemon server cleanly
+     * @private
+     * @memberof DaemonClient
+     */
+    private shutdown() {
+        // NOTE(Kelosky): this is not exposed yet, but will allow for a clean shut down if
+        // undocumented `--shutdown` is written to the persistent Processor.  It should be wrapped
+        // in a new header and handled in DaemonClient.ts, e.g. x-zowe-daemon-shutdown
+        Imperative.api.appLogger.debug("shutting down");
+        this.mClient.write(`Terminating server`);
+        this.mClient.end();
+        this.mServer.close();
+    }
+
+    /**
+     * Write data received from the daemon client to stdin.
+     * @param data Data to write to stdin
+     * @param expectedLength Expected length of the data to validate
+     * @private
+     * @memberof DaemonClient
+     */
+    private writeToStdin(data: string, expectedLength: number) {
+        // stdin.isTTY is checked by get-stdin and should be false if stdin contains valid data.
+        // We set it to false if the number of bytes received matches the expected `stdinLength`.
+        const stdin: any = process.stdin;
+        Object.defineProperty(process.stdin, "isTTY", { value: data.length !== expectedLength });
+
+        // Normally Node.js reads from process.stdin only once, and the stream is not reusable.
+        // To work around this, we store a copy of the `_readableState` object before stdin has been used.
+        // To reuse stdin, we reset the `_readableState` object back to its initial value.
+        if (stdin._readableStateOld == null) {
+            stdin._readableStateOld = { ...stdin._readableState };
+        } else {
+            stdin._readableState = { ...stdin._readableStateOld };
+            process.stdin.removeAllListeners();
+        }
+
+        // Calling stdin.write throws an EPIPE error because Node.js thinks the stream is read-only.
+        // So we simulate calls to stdin.write and stdin.end by pushing to the internal buffer.
+        process.stdin.push(data);
+        process.stdin.push(null);
+    }
+
     /**
      * Data event handler triggered for whenever data comes in on a connection
      * @private
      * @param {Buffer} data
      * @memberof DaemonClient
      */
     private data(data: Buffer) {
-
+        // Split JSON body and binary data from multipart response
         const stringData = data.toString();
+        const jsonEndIdx = stringData.indexOf("}\n");
```

Comment:
What if `}\n` is part of the data coming back ?
Maybe reading a JSON file from USS ?

Reply:
This method only parses data sent from client-to-server, which is limited to the properties listed above: argv, cwd, env, stdin, stdinLength.
As long as none of these values contain new-line characters, then `}\n` should never be present inside the JSON object.
Any data coming back to be written to stdout/stderr (like the contents of a USS file) are in the data that gets sent back from server-to-client which is not parsed here.
Perhaps I could change the separator to use a form-feed (`}\f`) rather than new-line if that seems more robust?

Reply:
It may be more consistent with the use of the form feed char to indicate termination in other areas of the daemon implementation 😋